src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx

```diff
@@ -154,10 +154,11 @@ namespace xclhwemhal2 {
     }
     if (buf_size < new_size)
     {
-      void *temp = buf;
-      buf = (void*) realloc(temp,new_size);
-      if (!buf) // prevent leak of original buf
-        free(temp);
+      void *result = realloc(buf, new_size);
```

Comment:
It looks like there are a lot of copy-paste in this code at the first place... :-(

Reply:
You mean in `shim.cxx` or in general in XRT?

Reply:
Yes, between all the different `sw_emu`, `hw_emu`,... in XRT. But this is not your fault. :-)